benchmark/ubench.c

```diff
@@ -339,4 +345,125 @@ UBENCH_EX(simple, scalable_pool_with_os_memory_provider) {
 }
 #endif /* (defined UMF_POOL_SCALABLE_ENABLED) */
 
+#if (defined UMF_BUILD_LIBUMF_POOL_DISJOINT &&                                 \
+     defined UMF_BUILD_LEVEL_ZERO_PROVIDER && defined UMF_BUILD_GPU_TESTS)
+static void do_ipc_get_put_benchmark(alloc_t *allocs, size_t num_allocs,
+                                     size_t repeats,
+                                     umf_ipc_handle_t *ipc_handles) {
+    for (size_t r = 0; r < repeats; ++r) {
+        for (size_t i = 0; i < num_allocs; ++i) {
+            size_t handle_size = 0;
+            umf_result_t res =
+                umfGetIPCHandle(allocs[i].ptr, &(ipc_handles[i]), &handle_size);
+            if (res != UMF_RESULT_SUCCESS) {
+                fprintf(stderr, "umfGetIPCHandle() failed\n");
+                exit(-1);
+            }
+        }
+
+        for (size_t i = 0; i < num_allocs; ++i) {
+            umf_result_t res = umfPutIPCHandle(ipc_handles[i]);
+            if (res != UMF_RESULT_SUCCESS) {
+                fprintf(stderr, "umfPutIPCHandle() failed\n");
+                exit(-1);
+            }
+        }
+    }
+}
+
+int create_level_zero_params(level_zero_memory_provider_params_t *params) {
+    uint32_t driver_idx = 0;
+    ze_driver_handle_t driver = NULL;
+    ze_context_handle_t context = NULL;
+    ze_device_handle_t device = NULL;
+
+    int ret = init_level_zero();
+    if (ret != 0) {
+        fprintf(stderr, "Failed to init Level 0!\n");
+        return ret;
+    }
+
+    ret = find_driver_with_gpu(&driver_idx, &driver);
+    if (ret || driver == NULL) {
+        fprintf(stderr, "Cannot find L0 driver with GPU device!\n");
+        return ret;
+    }
+
+    ret = create_context(driver, &context);
+    if (ret != 0) {
+        fprintf(stderr, "Failed to create L0 context!\n");
+        return ret;
+    }
+
+    ret = find_gpu_device(driver, &device);
+    if (ret || device == NULL) {
+        fprintf(stderr, "Cannot find GPU device!\n");
+        return ret;
+    }
+
+    params->level_zero_context_handle = context;
+    params->level_zero_device_handle = device;
+    params->memory_type = UMF_MEMORY_TYPE_DEVICE;
+
+    return ret;
+}
+
+UBENCH_EX(ipc, disjoint_pool_with_level_zero_provider) {
+    const size_t BUFFER_SIZE = 100;
+    const size_t N_BUFFERS = 1000;
+    level_zero_memory_provider_params_t level_zero_params;
+
+    int ret = create_level_zero_params(&level_zero_params);
+    if (ret != 0) {
+        exit(-1);
+    }
+
+    alloc_t *allocs = alloc_array(N_BUFFERS);
+    umf_ipc_handle_t *ipc_handles = calloc(N_BUFFERS, sizeof(umf_ipc_handle_t));
```

Comment:
add check for ipc_handles == NULL

Reply:
Done

Reply:
Actually, `alloc_array` just terminates the process with `exit(-1)`. I think it is OK for the benchmark.